### DISCUSSION TITLE
JWebAssembly support for UI

### DISCUSSION THREAD

**Discussion by houssemradhouane** [i-net-software/JWebAssembly#57] [Q&A]
Is it possible to take Swing components of Java and compile them into wasm?

**Comment by Horcrux7** [comment]
Currently never has implements the Graphics and Graphics2D interface for Webassembly. This can theoretically be realised via the Canvas or OpenGL API in JavaScript. This interfaces are essential for Swing component.

But also if this would be down. It does not make sense for me. There would be no mouse and keyboard events. If you want develop a UI for the browser then you must do it via the DOM.

See also the [FAQ](https://github.com/i-net-software/JWebAssembly/wiki/FAQ) and the Hello World sample ([live](https://i-net-software.github.io/JWebAssembly/samples/HelloWorld/HelloWorld.html)) ([source](https://github.com/i-net-software/JWebAssembly/blob/master/docs/samples/HelloWorld/HelloWorld.java)).
